src/librustdoc/html/render/print_item.rs

```diff
@@ -934,16 +934,17 @@ fn item_trait(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, t: &clean:
     let cache = &cloned_shared.cache;
     let mut extern_crates = FxIndexSet::default();
 
-    if !t.is_object_safe(cx.tcx()) {
+    if !t.is_dyn_compatible(cx.tcx()) {
+        // FIXME(dyn_compat_renaming): Update URL once the Reference is updated and hits stable.
         write_section_heading(
             w,
-            "Object Safety",
-            "object-safety",
+            "Dyn Compatibility",
+            "dyn-compatibility",
             None,
             &format!(
-                "<div class=\"object-safety-info\">This trait is <b>not</b> \
+                "<div class=\"dyn-compatibility-info\">This trait is <b>not</b> \
                 <a href=\"{base}/reference/items/traits.html#object-safety\">\
-                object safe</a>.</div>",
+                dyn compatible</a> (formerly known as object safe).</div>",
```

Comment:
Let me know if you know of a 'better way' to present the "formerly known" part. For context, in the compiler & library docs, I went for MarkDown footnotes. I wonder if we should quote or italicize "object safe".